synapse/storage/databases/main/account_data.py

```diff
@@ -476,7 +476,7 @@ async def add_account_data_for_user(
             self._account_data_stream_cache.entity_has_changed(user_id, next_id)
             self.get_account_data_for_user.invalidate((user_id,))
             self.get_global_account_data_by_type_for_user.invalidate(
-                (account_data_type, user_id)
+                (user_id, account_data_type)
             )
```

Comment:
The room case of this seems to prefill instead of invalidate, weirdly. I think this change is fine though, just noting a discrepancy:

https://github.com/matrix-org/synapse/blob/3e0536cd2afb5a640619bd872fc27b068ec3eb9b/synapse/storage/databases/main/account_data.py#L444-L446

Reply:
It is a little bit odd, but it's correct either way — maybe it was an intentional optimisation, maybe it's just inconsistent.

Reply:
I think prefilling would be a bit of an optimization, but probably not worth it. 🤷